feat: add C implementation for `stats/base/dists/uniform/mgf`

0PrashantYadav0 opened the pull request:
Resolves #3832

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for @stdlib/stats/base/dists/uniform/mgf along with relevant tests, benchmarks and examples.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3832

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

0PrashantYadav0 committed:
feat: add C implementation for uniform mgf

0PrashantYadav0 committed:
fix: added example in README.md

0PrashantYadav0 committed:
fix: removed unwanted examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

stdlib-bot committed:
chore: update copyright years

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/mgf/README.md:
```diff
@@ -155,6 +155,96 @@ for ( i = 0; i < 10; i++ ) {
+
+#### stdlib_base_dists_uniform_mgf( t, a, b )
+
+Evaluating the logarithm of the [moment-generating function][mgf] of a [uniform][uniform-distribution] distribution with parameters `a` (minimum support) and `b` (maximum support).
```
```suggestion
Evaluates the [moment-generating function][mgf] (MGF) for a [continuous uniform][uniform-distribution] distribution with parameters `a` (minimum support) and `b` (maximum support).
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/mgf/benchmark/benchmark.js:
```diff
@@ -30,18 +31,26 @@ var mgf = require( './../lib' );
+	min = new Float64Array( len );
+	max = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		t[ i ] = ( randu() * 20.0 ) - 10.0;
```
Would you be able to update the logic to generate values to our latest conventions, as exemplified in e.g. PR https://github.com/stdlib-js/stdlib/pull/4929?

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/mgf/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,73 @@
+	min = new Float64Array( len );
+	max = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		t[ i ] = ( randu() * 20.0 ) - 10.0;
```
Would you be able to update the logic to generate values to our latest conventions, like we now have in the `benchmark.js` file in this PR after merging in the latest changes?

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/mgf/src/main.c:
```diff
@@ -0,0 +1,50 @@
+* @return    evaluated MGF
+*
+* @example
+* double y = mstdlib_base_dists_uniform_mgfgf( 2.0, 0.0, 4.0 );
+* // returns ~372.495
```
```suggestion
* double y = stdlib_base_dists_uniform_mgf( 2.0, 0.0, 4.0 );
* // returns ~372.495
```

Planeshifter reviewed (changes requested):
Thanks, @0PrashantYadav0!

A few small fixes and a request to please update the benchmark files to match latest conventions; otherwise, this one should be ready to land!

Planeshifter committed:
Merge branch 'develop' into uniform/mgf

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/uniform/mgf/benchmark/benchmark.js:
```diff
@@ -31,6 +31,7 @@ var mgf = require( './../lib' );
 // MAIN //
 
 bench( pkg, function benchmark( b ) {
+	var len;
```
```suggestion
```

Planeshifter committed:
bench: remove duplicated declaration

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

0PrashantYadav0 commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into uniform/mgf

0PrashantYadav0 committed:
fix: changes according to given code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Planeshifter committed:
chore: fix section markup

Planeshifter reviewed (approved):
Thanks, @0PrashantYadav0!
